bin/p2-rm/rm.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/consulutil"
+	"github.com/square/p2/pkg/kp/podstore"
+	"github.com/square/p2/pkg/kp/rcstore"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/rc"
+	"github.com/square/p2/pkg/rc/fields"
+	"github.com/square/p2/pkg/types"
+)
+
+type store interface {
+	NewSession(name string, renewalCh <-chan time.Time) (kp.Session, chan error, error)
+	DeletePod(podPrefix kp.PodPrefix, nodename types.NodeName, podId types.PodID) (time.Duration, error)
+}
+
+type P2RM struct {
+	Store    store
+	RCStore  rcstore.Store
+	Client   consulutil.ConsulClient
+	Labeler  labels.ApplicatorWithoutWatches
+	PodStore podstore.Store
+
+	LabelID      string
+	NodeName     types.NodeName
+	PodName      types.PodID
+	PodUniqueKey types.PodUniqueKey
+}
+
+// NewLegacyP2RM is a constructor for the P2RM type which configures it to
+// remove a "legacy" pod. It will generate the storage types based on its
+// api.Client argument
+func NewLegacyP2RM(client consulutil.ConsulClient, podName types.PodID, nodeName types.NodeName, labeler labels.ApplicatorWithoutWatches) *P2RM {
+	rm := &P2RM{}
+	rm.LabelID = path.Join(nodeName.String(), podName.String())
+	rm.PodName = podName
+	rm.NodeName = nodeName
+	rm.PodUniqueKey = ""
+	rm.configureStorage(client, labeler)
+	return rm
+}
+
+// Constructs a *P2RM configured to remove a pod identified by a PodUniqueKey (uuid)
+func NewUUIDP2RM(client consulutil.ConsulClient, podUniqueKey types.PodUniqueKey, podID types.PodID, labeler labels.ApplicatorWithoutWatches) *P2RM {
+	rm := &P2RM{}
+	rm.LabelID = podUniqueKey.String()
+	rm.PodName = podID
+	rm.NodeName = "" // don't need node name to look up a uuid pod
+	rm.PodUniqueKey = podUniqueKey
+	rm.configureStorage(client, labeler)
+	return rm
+}
+
+func (rm *P2RM) configureStorage(client consulutil.ConsulClient, labeler labels.ApplicatorWithoutWatches) {
+	rm.Client = client
+	rm.Store = kp.NewConsulStore(client)
+	rm.RCStore = rcstore.NewConsul(client, labeler, 5)
+	rm.Labeler = labeler
+	rm.PodStore = podstore.NewConsul(client.KV())
+}
+
+func (rm *P2RM) checkForManagingReplicationController() (bool, fields.ID, error) {
+	podLabels, err := rm.Labeler.GetLabels(labels.POD, rm.LabelID)
+	if err != nil {
+		return false, "", fmt.Errorf("unable to check node for labels: %v", err)
+	}
+
+	if podLabels.Labels.Has(rc.RCIDLabel) {
+		return true, fields.ID(podLabels.Labels.Get(rc.RCIDLabel)), nil
+	}
+
+	return false, "", nil
+}
+
+func (rm *P2RM) decrementDesiredCount(id fields.ID) error {
+	session, _, err := rm.Store.NewSession(sessionName(id), nil)
```

Comment:
ha i have no memory of doing that, but i stand by the decision to remove it since it's confusing and doesn't change behavior.